### ISSUE TITLE
"Algorithm negotiation fail caused by git@github.com:bazelbuild/bazel-skylib"

### ISSUE THREAD

**Issue by yifan-gu** [bazelbuild/bazel#4762]
### Description of the problem / feature request:

My `bazel build` returns the following error:
```
ERROR: Skipping 'tarball': error loading package '': Encountered error while reading extension file 'lib.bzl': no such package '@bazel_skylib//': Error cloning repository: git@github.com:bazelbuild/bazel-skylib: Algorithm negotiation fail caused by git@github.com:bazelbuild/bazel-skylib: Algorithm negotiation fail caused by Algorithm negotiation fail
WARNING: Target pattern parsing failed.
ERROR: error loading package '': Encountered error while reading extension file 'lib.bzl': no such package '@bazel_skylib//': Error cloning repository: git@github.com:bazelbuild/bazel-skylib: Algorithm negotiation fail caused by git@github.com:bazelbuild/bazel-skylib: Algorithm negotiation fail caused by Algorithm negotiation fail
INFO: Elapsed time: 0.050s
FAILED: Build did NOT complete successfully (0 packages loaded)
    currently loading: 
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

I was running `bazel build tarball` under the root dir of the [tectonic installer](https://github.com/coreos/tectonic-installer) probject, and hit this error. But it seems that the error only happens on some particular distro/setup. Doesn't reproduce on other people's machine.

### What operating system are you running Bazel on?

Ubuntu 16.04
```
$ uname -a
Linux yifan-coreos 4.2.0-42-generic #49-Ubuntu SMP Tue Jun 28 21:26:26 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

### What's the output of `bazel info release`?

```
release 0.11.0
```

###  Have you found anything relevant by searching the web?

Not really

### Any other information, logs, or outputs that you want to share?

```
$ git version
git version 2.7.4
```
Besides, I can definitely `git clone git@github.com:bazelbuild/bazel-skylib.git` with no problem.

**Comment by davido** [comment]
JGit version is stale in Bazel distribution: v4.0.1.201506240215-r.
This version uses stale version of JSch library: 0.1.51.

This library only contains weak cryptographic algorithms, that was recently removed by GitHub. See this announcement: [1].

Older gerrit versions were affected by this problem as well, see this discussion: [2].

* [1] https://githubengineering.com/crypto-removal-notice
* [2] https://groups.google.com/d/topic/repo-discuss/AMVDwTIeDJw/discussion

**Comment by davido** [comment]
Should be fixed by: [1].

* [1] https://githubengineering.com/crypto-removal-notice

**Comment by laszlocsomor** [comment]
@davido : Thanks for commenting and for your commit! Is it ready for review?

**Comment by davido** [comment]
@laszlocsomor I wasn't able to test it in my dev environment and was hoping to get verification feedback from Bazel CI against gerrit CL. Unfortunately the verification firehose doesn't seem to work?

**Comment by laszlocsomor** [comment]
Yes, we've been replacing our Jenkins-based CI with Buildkite -- [details are here](https://groups.google.com/d/msg/bazel-dev/dQ6_Twlm9Yc/vf9YC1FPDAAJ).

Could you please create a PR instead of a Gerrit code review? Since Buildkite doesn't support Gerrit, we can no longer trigger presubmits from Gerrit reviews, only from GitHub PRs.

**Comment by davido** [comment]
> Could you please create a PR instead of a Gerrit code review?

Seriously, you are telling that me - Gerrit Code Review maintainer -  to stop using Gerrit and prefer GitHub instead? That pretty much sucks.

**Comment by laszlocsomor** [comment]
I'm sorry about that.

I also like the Gerrit UI more than the GitHub UI, alas it's not my decision.

**Comment by laszlocsomor** [comment]
Also, AFAIK there's no Buildkite integration for Gerrit. What can we do?

**Comment by davido** [comment]
> Also, AFAIK there's no Buildkite integration for Gerrit.

It shouldn't be hard to create Gerrit integration with any system. But that's the problem for another day. I will upload GitHub PR later today.

**Comment by yifan-gu** [comment]
@davido Thank you for the PR! So make my build successful without waiting for the PR and release, what should I do? Should I downgrade the bazel or build a new bazel from the the master?

**Comment by davido** [comment]
@yifan-gu Two libraries used by Bazel are outdated, so obviously, it wouldn't help to downgrade. Just apply my CL (fetch it from Gerrit) and build custom Bazel version from master my CL applied should fix your problem.

**Comment by yifan-gu** [comment]
@davido Thank you, let me try.

**Comment by davido** [comment]
Here is the link to cherry-pick the CL: "git fetch https://bazel.googlesource.com/bazel refs/changes/10/43610/2 && git cherry-pick FETCH_HEAD".

**Comment by lukeweber** [comment]
Any ideas on when this will be released?

**Comment by davido** [comment]
@lukeweber The fix is included in March Bazel release: #4582. I could already tesr rc1 of this upcoming release, see this announcement: https://groups.google.com/d/topic/bazel-discuss/Z5BH_SEjvDY/discussion.
